x-pack/plugins/observability_solution/infra/public/pages/metrics/inventory_view/components/table_view.tsx

```diff
@@ -69,14 +81,14 @@ export const TableView = (props: Props) => {
           <EuiToolTip content={tooltipText}>
             <EuiButtonEmpty
               data-test-subj="infraColumnsButton"
-              onClick={() => setOpenPopoverId(uniqueID)}
+              onClick={() => toggleAssetPopover(uniqueID)}
```

Comment:
I think here we can pass the `item.node.id` directly so you don't have to do the split of the `uniqueID`, wdyt? My concern is that the `uniqueID` is using the item.node.name] which is not always the same as `item.node.id` (in case of host it is but it's not in case of containers for example)

Reply:
There is this [comment](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/public/pages/metrics/inventory_view/components/table_view.tsx#L64) which tells us why we are using this `uniqueID`, and to avoid breaking anything I didn't change it, but if you think `item.node.id` will work I can change it

Reply:
Yes it is needed for the table - what I am talking about is in
`setFlyoutUrlState({ detailsItemId: nodeId, assetType: nodeType });` not `setOpenPopoverId(uniqueID);`
So what I am suggesting is something like: 
```
  const toggleAssetPopover = (uniqueID: string, nodeId: string) => {
    if (isFlyoutMode) {
      setFlyoutUrlState({ detailsItemId: nodeId, assetType: nodeType });
    } else {
      setOpenPopoverId(uniqueID);
    }
  };
  ```
  And then: 
  ```
              onClick={() => toggleAssetPopover(uniqueID, item.node.id)}
```

Reply:
will do that then!

Reply:
Awesome 💯 Thank you! And can you please check with containers as well (Opening the flyout and then clicking on `Open as page` - and checking the URL for the `detailsItemId `)? 
Test Dtata with Synthtrace: `node scripts/synthtrace infra_docker_containers.ts --live`

Reply:
working fine! 
![image](https://github.com/user-attachments/assets/70e1aef1-a057-43d3-ae55-5e28a7674734)